src/classes/CRLP_Rollup_SEL.cls

```diff
@@ -165,9 +183,45 @@ public class CRLP_Rollup_SEL {
      * @return requested rollup__mdt
      */
     public static Rollup__mdt getRollupById(Id id) {
-
         return getRollupsById().get(id);
+    }
+
+    /**
+     * @description Blacklist of
+     */
+    @TestVisible
+    private static final Map<SObjectField, Set<String>> UNSUPPORTED_OPERATIONS_BY_ROLLUP_FIELD = new Map<SObjectField, Set<String>> {
+        Schema.Rollup__mdt.Date_Field__c => new Set<String> {}, // All Operations Supported
+        Schema.Rollup__mdt.Amount_Field__c => new Set<String> {
+            CRLP_Operation.RollupType.Count.name(), 
+            CRLP_Operation.RollupType.First.name(), 
+            CRLP_Operation.RollupType.Last.name()
+        }, 
+        Schema.Rollup__mdt.Detail_Field__c => new Set<String> {
+            CRLP_Operation.RollupType.Sum.name(), 
+            CRLP_Operation.RollupType.Count.name(), 
+            CRLP_Operation.RollupType.Average.name(), 
+            CRLP_Operation.RollupType.Years_Donated.name(), 
+            CRLP_Operation.RollupType.Donor_Streak.name(), 
+            CRLP_Operation.RollupType.Best_Year.name(), 
+            CRLP_Operation.RollupType.Best_Year_Total.name()
+        }
+    };
 
+    /**
+     * @description Determines if specific Rollup.Amount, Date, or Detail__c field should be included in a query
+     * of the Detail object based on the specific Rollup operation type. In the event that the field refers to a
+     * field that will never be used in the rollup logic, this prevents that field from being included in the Query.
+     * @param  rollup Rollup__mdt
+     * @param  metadataRelationship SObjectField of Rollup__mdt Metadata Relationship (Amount, Detail or Date field)
+     * @return True if the rollup Operation is not null and not in the UNSUPPORTED_OPERATIONS_BY_ROLLUP_FIELD map
+     */
+    @TestVisible
+    private static Boolean isRollupFieldNeededInQuery(Rollup__mdt rollup, Schema.SObjectField metadataRelationship) {
+        return
+            rollup.Operation__c != null
+            && rollup.get(metadataRelationship) != null
```

Comment:
@spelak-salesforce I added this additional condition to the method so that it can prevent scenarios where the Detail__c, Amount__c or Data__c field is null. It's not typical, but it's not impossible. I also renamed the method because I felt that `isRollupFieldNeededInQuery` more accurately defined the usage.